pkg/config/common.go

```diff
@@ -102,19 +102,15 @@ func LoadConfiguration() (*configuration.Configuration, error) {
 		}
 	}
 
-	tokenListJson := os.Getenv(TokenListEnv)
-	if file, err := os.ReadFile(tokenListJson); err == nil {
-		// if the envvar points to a file, read it; otherwise the envvar contents is expected to be JSON
-		tokenListJson = string(file)
-	}
-	if tokenListJson == "" {
-		tokenListJson = "[]"
+	// Graceful tokenlist unmarshalling
+	tokenListJsonFilename := os.Getenv(TokenListEnv)
+	tokenListJsonFile, err := ReadTokenConfig(tokenListJsonFilename)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse token list %s: %w", tokenListJsonFilename, err)
 	}
-
-	var payload []configuration.Token
-	err = json.Unmarshal([]byte(tokenListJson), &payload)
+	tokenWhiteList, err := UnmarshalTokenConfig([]byte(tokenListJsonFile))
```

Comment:
the `networks` input argument should be the `NetworkEnv`.

Reply:
It works either way as an empty `networks` is interpreted as no filter. In that case, we can simplify the code and remove that parameter from `UnmarshalTokenConfig`.